### PR TITLE
Implement HAS random buff from #1085

### PR DESCRIPTION
**Pull request by sonohoshi**
This PR has changes of #1085.

Please Check CI and result of unit test.

### CODE REVIEW

**Review by ipdae — CHANGES_REQUESTED** [review]
we must add test case for HackAndSlash execute with crystal skill

**Comment by ipdae** [comment]
test failed

File: .Lib9c.Tests/Action/HackAndSlashTest.cs [hunk 1]
```diff
@@ -1018,6 +1016,146 @@ public void CheckRewardItems(bool backward, int worldId, int stageId)
...
+            {
+                Assert.Equal(stageId, skillState.StageId);
+                Assert.Equal(
+                    hasCrystalSkill
+                        ? _tableSheets.CrystalStageBuffGachaSheet[stageId].MaxStar
+                        : log.clearedWaveNumber,
+                    skillState.StarCount);
+            }
+
+            Assert.Empty(skillState.SkillIds);
```
Comment (ipdae) [inline .Lib9c.Tests/Action/HackAndSlashTest.cs:1156]: it should be check also clear case.

File: .Lib9c.Tests/Action/HackAndSlashTest.cs [hunk 2]
```diff
@@ -1018,6 +1016,146 @@ public void CheckRewardItems(bool backward, int worldId, int stageId)
...
+                stageId,
+                _tableSheets.GetStageSimulatorSheets(),
+                _tableSheets.CostumeStatSheet,
+                StageSimulator.ConstructorVersionV100080);
+            simulator.Simulate(1);
+            var log = simulator.Log;
+            var skillStateIValue =
+                nextState.GetState(skillStateAddress);
+            var serialized = skillStateIValue as List;
+            Assert.NotNull(serialized);
```
[the remark this answers is not in the document]
Comment (sonohoshi) [reply .Lib9c.Tests/Action/HackAndSlashTest.cs:1137]: Yes. client is already handling about null.

File: .Lib9c.Tests/Action/HackAndSlashTest.cs (hunk 1 above)
Comment (sonohoshi) [reply .Lib9c.Tests/Action/HackAndSlashTest.cs:1156]: It already check clear case and failed case.

**Review by ipdae — APPROVED** [review]

**Pull request merged**